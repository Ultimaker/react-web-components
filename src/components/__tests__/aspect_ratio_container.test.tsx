// Copyright (c) 2018 Ultimaker B.V.
import * as React from 'react';
import { shallow } from 'enzyme';

// component
import AspectRatioContainer from '../aspect_ratio_container';

describe('The Image component', () => {
    let props;
    let wrapper;

    beforeEach(() => {
        props = {
            aspectRatio: 4 / 3,
            id: 'container-id',
            className: 'container-class',
        };
        wrapper = shallow(<AspectRatioContainer {...props}><img id="myPrettyPicture" className="test-image" src="http://ultimaker.com" alt="a test image" /></AspectRatioContainer>);
    });

    it('should render', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
