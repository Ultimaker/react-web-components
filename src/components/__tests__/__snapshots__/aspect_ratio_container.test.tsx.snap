// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Image component should render 1`] = `
<div
  className="aspect-ratio-container container-class"
  id="container-id"
>
  <div
    className="aspect-ratio-container__height-setter"
    style={
      Object {
        "paddingBottom": "75%",
      }
    }
  >
    <img
      alt="a test image"
      className="aspect-ratio-container__content test-image"
      id="myPrettyPicture"
      src="http://ultimaker.com"
    />
  </div>
</div>
`;
